### PR TITLE
Prepare for publishing the Atom package

### PR DESCRIPTION
**Pull request by MetRonnie**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,45 @@
...
+- Cylc 7
+- Cylc 8
+- ISO 8601 datetimes
+- Jinja2 (can be used with a Jinja extension e.g. [atom-jinja2](https://atom.io/packages/atom-jinja2))
+
+Check [here](https://github.com/cylc/cylc-flow/issues/2752) for info on supported syntax features.
+
+## Installation
+
+Currently, the only way to install is to clone the repository into your `.atom/` directory:
```
Comment (wxtim) [inline README.md]: Is this true if you are pushing it to the atom package repo?

**Review by wxtim — APPROVED** [review]
I'm happy with this as an interim, although it's not highlighting
``[[subsections]]`` correctly in my copy of atom - still quite handy.
Checking the basic syntax highlighting can be another ticket.

File: README.md (hunk 1 above)
Comment (MetRonnie) [reply README.md]: Just updated the instructions

**Comment by MetRonnie** [comment]
@wxtim Did you set the file extensions in the config? If not I suspect that's the language-ini package being used instead. Unfortunately we haven't set the file extensions in cylc-textmate-grammar because `.rc` might be too broad. But if everyone sets it themselves manually anyway, might as well we just set it in cylc-textmate-grammar?

**Comment by wxtim** [comment]
foirr

**Comment by MetRonnie** [comment]
Why does it say "cylc rc file"? It should just say "cylc"

**Comment by wxtim** [comment]
Ignore me - I've just cleaned out `.atom` folder and everything is now much better

**Comment by MetRonnie** [comment]
I've added a package version bump (might as well include it here) to 0.1.0 - not convinced about 1.0.0 until the extension works out of the box without having to edit the Atom config

File: package.json [hunk 2]
```diff
@@ -1,12 +1,10 @@
 {
     "name": "language-cylc",
-    "version": "0.0.0",
+    "version": "0.1.0",
```
Comment (oliver-sanders) [inline package.json]: This should be stable so perhaps we should go straight to `1.0.0` so it doesn't look like a flaky alpha release.

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  deploy:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+    - name: Install apm (with Atom)
+      env:
+        TEMP_DEB: atom-amd64.deb
+      run: |
+        wget -nv -O "$TEMP_DEB" https://atom.io/download/deb &&
+        sudo apt -qq install "$TEMP_DEB"
```
Comment (oliver-sanders) [inline .github/workflows/deploy.yml:18]: Out of interest does Atom not register their package with debian for direct installation?

**Review by oliver-sanders — APPROVED** [review]
I've not tested it but I happy that the two of you have.

Next step release and see if we get any bug reports. I might be worth announcing this to users and requesting feedback (people will often just suffer bugs rather than report them).

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (MetRonnie) [reply .github/workflows/deploy.yml:18]: I don't think so? There's another method they give here first: https://flight-manual.atom.io/getting-started/sections/installing-atom/#debian-and-ubuntu-debapt involving "adding their official
package repository to your system" but it looks more complicated to me

File: package.json (hunk 2 above)
Comment (MetRonnie) [reply package.json]: My reasoning was that currently the user has to manually set the `.rc` file association, so it feels a bit alpha-y

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (oliver-sanders) [reply .github/workflows/deploy.yml:18]: Strange choice for Atom but fair enough.

File: package.json (hunk 2 above)
Comment (oliver-sanders) [reply package.json]: Aaah, yes, that it is a bit alpha-y, fair enough.

What was the barrier to defining file associations in the TextMate grammar itself? Can't remember.

Comment (MetRonnie) [reply package.json]: I think there was a worry `.rc` is too broad as there are other configuration files that use it

Comment (oliver-sanders) [reply package.json]: Ah yes, TextMate grammars don't support the more advanced filename associations we require.

**Comment by MetRonnie** [comment]
Through trial & error, I have found out we can use
```json
{
    "fileTypes": ["suite.rc"]
}
```
in `cylc.tmLanguage.json` (cylc-textmate-grammar).

This is ignored by VSCode, which has its own file association rules.

**Comment by wxtim** [comment]
Another win from Ronnie!

> `"fileTypes": ["suite.rc"]`

We probably want to add `flow.rc` & `flow-test.rc` if this is a list.

Having said which, perhaps we should just merge.

Don't forget to announce on Yammer and Discourse, solicit feedback.

**Comment by oliver-sanders** [comment]
The [TextMate documentation](https://macromates.com/manual/en/language_grammars) is wonderfully vague about what you can do with the `fileTypes` field:

> this is an array of file type extensions that the grammar should (by default) be used with

Presumably we can't have globs here?

Will work nicely when we've moved to the `.cylc` extension.

**Comment by MetRonnie** [comment]
Globs don't work unfortunately. @wxtim Is it `flow-test.rc` or `flow-tests.rc`?

**Comment by oliver-sanders** [comment]
> Globs don't work unfortunately

Never mind!

It's `flow-tests.rc`

**Comment by MetRonnie** [comment]
Tested the Cylc.tmBundle in Pycharm and it works!

**Review by oliver-sanders — APPROVED** [review]
Tried and tested with atom.

File associations worked after I patched the textmate grammar to the latest commit.

**Pull request merged**